### PR TITLE
FileInfo tests for mzid validation

### PR DESCRIPTION
**Pull request by lars20070**
Two new `FileInfo` TOPP tool tests for the validation of `mzid`.

### CODE REVIEW

**Comment by timosachsenberg** [comment]
+1 on pass

File: src/tests/topp/CMakeLists.txt [hunk 1]
```diff
@@ -557,6 +557,10 @@ set_tests_properties("TOPP_FileInfo_11" PROPERTIES WILL_FAIL 1) ## the mzML has
 add_test("TOPP_FileInfo_12" ${TOPP_BIN_PATH}/FileInfo -in ${DATA_DIR_TOPP}/FileInfo_12_input.mzML -i  -no_progress)
 set_tests_properties("TOPP_FileInfo_12" PROPERTIES WILL_FAIL 0) ## this mzML has an index, should succeed
 add_test("TOPP_FileInfo_13" ${TOPP_BIN_PATH}/FileInfo -in ${DATA_DIR_TOPP}/FileInfo_13_input.consensusXML -no_progress) # empty file should not crash
+add_test("TOPP_FileInfo_14" ${TOPP_BIN_PATH}/FileInfo -in ${DATA_DIR_TOPP}/FileInfo_14_input.mzid -v -no_progress -out FileInfo_14.tmp)
+add_test("TOPP_FileInfo_14_out1" ${DIFF} -whitelist "File name" -in1 FileInfo_14.tmp -in2 ${DATA_DIR_TOPP}/FileInfo_14_output.txt )
+add_test("TOPP_FileInfo_15" ${TOPP_BIN_PATH}/FileInfo -in ${DATA_DIR_TOPP}/FileInfo_15_input.mzid -v -no_progress -out FileInfo_15.tmp)
+add_test("TOPP_FileInfo_15_out1" ${DIFF} -whitelist "File name" -in1 FileInfo_15.tmp -in2 ${DATA_DIR_TOPP}/FileInfo_15_output.txt )
```
Comment (lars20070) [inline src/tests/topp/CMakeLists.txt]: A second whitelist entry is required for row 8.
https://github.com/OpenMS/OpenMS/pull/1851/files#diff-1e0115b434e78c7a5f2819804fcaac50R8
```
... -whitelist "File name" -whitelist "Validation Error" ...
```
Is that the notation?

Comment (timosachsenberg) [inline src/tests/topp/CMakeLists.txt]: Its case sensitive.
I can't recall but there are other tests with multiple whitelists.

Comment (timosachsenberg) [inline src/tests/topp/CMakeLists.txt]: I thing you need to remove the second -whitelist

**Comment by lars20070** [comment]
@timosachsenberg The test still fails. I haven't figured out how to whitelist both file names here.
https://github.com/OpenMS/OpenMS/pull/1851/files#diff-1e0115b434e78c7a5f2819804fcaac50R4

**Comment by timosachsenberg** [comment]
travis is still running... is there already a CDASH entry that shows the error?

**Comment by lars20070** [comment]
The last `FileInfo` test failed. - How do I exclude both lines 4 and 8 from the comparison?
https://github.com/OpenMS/OpenMS/pull/1851/files#diff-5109b29109edd5406fa6b7cf671e57f6R563

**Comment by cbielow** [comment]
http://cdash.openms.de/testDetails.php?test=30863291&build=167275

**Comment by cbielow** [comment]
using
``` ...  -whitelist "File name" "line 327 column 183" ... ```

should do it.

(right now you are using ```Validation Error``` (which is != "Validation error", upper/lower case issue), and also seems too generic (since we want other validation errors to show up -- given that they do not mention the path + filename.


Another (better?!) solution: change the error message and only print the filename (without path).
This avoids whitelisting altogether and should be specific enough (since line number is mentioned).

**Comment by lars20070** [comment]
Thanks @cbielow. Now I understand. Any line containing this substring is skipped.
BTW, why `-whitelist`? These lines are blacklisted.

**Comment by hendrikweisser** [comment]
> These lines are blacklisted.

They are whitelisted for the test - they always pass. Makes sense to me.

**Comment by lars20070** [comment]
Ok.

**Pull request merged**